### DISCUSSION TITLE
when current restore excludes pv/pvc, pod no-pvc volume will skip?

### DISCUSSION THREAD

**Discussion by BAMSHK** [vmware-tanzu/velero#5481] [Community support Q&A]
![image](https://user-images.githubusercontent.com/40750078/197498842-c31c2e23-bfb5-4601-afb1-0f734f7b0e22.png)
when current restore excludes pv/pvc, pod no-pvc volume will skip?

**Comment by sseago** [comment]
If a user is explicitly *not* restoring PVCs or PVs, then we don't create the podvolumerestore, since the whole purpose of the podvolumerestore is to restore the content of the PV/PVC, which has been excluded.

**Comment by BAMSHK** [comment]
Does pvcs or PV include all of the above types of volume?


------------------&nbsp;Original&nbsp;------------------
From: &nbsp;"Scott ***@***.***&gt;;
Send time:&nbsp;Monday, Oct 24, 2022 11:45 PM
***@***.***&gt;; 
***@***.***&gt;; ***@***.***&gt;; 
Subject: &nbsp;Re: [vmware-tanzu/velero] when current restore excludes pv/pvc, pod no-pvc volume will skip? (Discussion #5481)





 
If a user is explicitly not restoring PVCs or PVs, then we don't create the podvolumerestore, since the whole purpose of the podvolumerestore is to restore the content of the PV/PVC, which has been excluded.

**Comment by sseago** [comment]
PVCs and PVs are PersistentVolumeClaims and PersistentVolumes. All types of persistent volumes are modeled by these resource types. The code referenced above is specifically related to PodVolumeRestores. A PodVolumeRestore represents a specific PV/PVC pair, as mounted by a specific Pod which will be restored. If we're excluding PVs and/or PVCs, there would be nothing for Restic/Kopia to do, since the volume to be restored is explicitly excluded by the velero restore.

**Reply by BAMSHK** [reply]
oh，pv not only k8s PersistentVolumes Resource, it include pod volume?
![Uploading image.png…]()

**Reply by BAMSHK** [reply]
![image](https://user-images.githubusercontent.com/40750078/197797390-8edb651e-f40a-4ae2-a3aa-f4cc2837cfd9.png)

**Reply by sseago** [reply]
PV is the kubernetes PersistentVolume resource, which is referenced by the pod -- but if we're excluding the PVs and PVCs, then the underlying PV (whether it's azure, ceph, cinder, or whatever) won't be created, so there won't be a mounted volume for Restic to copy data to.
